nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py

```diff
@@ -615,6 +620,12 @@ def __init__(
                     preserve_frame_confidence=preserve_frame_confidence,
                     confidence_method_cfg=confidence_method_cfg,
                 )
+            elif use_cuda_graph_decoder:
+                from nemo.collections.asr.parts.submodules.cuda_graph_rnnt_greedy_decoding import (
+                    RNNTGreedyDecodeCudaGraph,
+                )
+
+                self._greedy_decode = RNNTGreedyDecodeCudaGraph(max_symbols_per_step, torch.device("cuda"), self)
```

Comment:
Is it possible to avoid hardcoding first cuda device here?
The problem is that I get an error when trying to transcribe on the other device (I have 2 GPUs on my local machine):

```shell
python examples/asr/speech_to_text_eval.py \
 pretrained_name=stt_en_fastconformer_transducer_large \
 dataset_manifest=~/code/asr_particles/manifests/librispeech/test_other.json \
 batch_size=16 \
 output_filename=test_other_decoded.jsonl \
 amp=false \
 amp_dtype=bfloat16 \
 ++rnnt_decoding.greedy.use_cuda_graph_decoder=true \
 rnnt_decoding.greedy.loop_labels=false \
 cuda=1
```
fails with "RuntimeError: Capture must end on the same stream it began on."

Reply:
I'm looking into it. https://pytorch.org/docs/stable/tensor_attributes.html#torch.device says:

> If the device ordinal is not present, this object will always represent the current device for the device type, even after [torch.cuda.set_device()](https://pytorch.org/docs/stable/generated/torch.cuda.set_device.html#torch.cuda.set_device) is called; e.g., a [torch.Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor) constructed with device 'cuda' is equivalent to 'cuda:X' where X is the result of [torch.cuda.current_device()](https://pytorch.org/docs/stable/generated/torch.cuda.current_device.html#torch.cuda.current_device).

Maybe the current device is being changed at some point inside nemo...

Reply:
Okay, it's a little bit complicated...

https://github.com/NVIDIA/NeMo/issues/8436#issue-2137592510

I believe I have a workaround to get this through, though.

Reply:
Should be fixed after https://github.com/NVIDIA/NeMo/pull/8191/commits/2c2da1a77f0cbb1f6b8a79c31e1fe20d417fe953